newIDE/app/src/Utils/GDevelopServices/Authentication.js

```diff
@@ -141,8 +144,25 @@ export default class Authentication {
 
   getFirebaseUser = (cb: (any, ?FirebaseUser) => void) => {
     if (!this.isAuthenticated()) return cb({ unauthenticated: true });
+    this.auth.currentUser.reload().then(() => cb(null, this.firebaseUser));
+  };
 
-    cb(null, this.firebaseUser);
+  sendFirebaseEmailVerification = (cb: () => void): Promise<void> => {
+    return this.auth.currentUser.reload().then(() => {
+      if (this.firebaseUser && !this.firebaseUser.emailVerified) {
+        sendEmailVerification(this.auth.currentUser)
+          .then(() => {
+            cb();
+          })
+          .catch((error: Error) => {
+            showErrorBox({
```

Comment:
Nit: prefer `then(() => { cb(); }, (error: Error) => { ... })`; in most cases, because the catch will swallow errors that are launched by cb (making debugging of an unrelated error in your callback difficult).
With the `then(cb, errorCb)` form, the error callback is only called for the sendEmailVerification related errors - which is good